### PR TITLE
feat: add publish image to ghcr pipeline

### PR DESCRIPTION
**Pull request by matthias-michel**

### CODE REVIEW

**Comment by s4heid** [comment]
@matthias-michel reuse is complaining about the`.dockerignore`. This should also be added to the [config file](https://github.com/SAP/neonbee/blob/main/.reuse/dep5).

File: docs/docker-image.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Docker Image
+
+## Pull the image
+
```
Comment (dapeng-wang) [inline docs/docker-image.md:4]: Please add a section explaining the docker login process.

File: Dockerfile [hunk 2]
```diff
@@ -1,20 +1,31 @@
...
+RUN gradle -x test -x spotlessJava \
+    -x spotlessCheck -x spotlessJavaCheck \
+    -x spotbugsMain -x spotbugsTest \
+    -x pmdMain -x pmdTest \
+    -x checkstyleMain -x checkstyleTest \
+    -x violations \
+    -x reportCoverage \
+    -x testJar -x testSourcesJar \
+    -x javadoc -x javadocJar -x testJavadoc -x testJavadocJar \
+    --no-daemon clean build
```
Comment (dapeng-wang) [inline Dockerfile:15]: The gradle build steps hangs on my Mac M1.

 nerdctl build -t neonbee-core . --platform=linux/amd64

**Review by dapeng-wang — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (matthias-michel) [reply Dockerfile:15]: Seems to be for every environment individual. It seems to work for the most people and also (most importantly) for the GitHub Workflow pipeline building the release package.

**Review by dapeng-wang — APPROVED** [review]

**Pull request merged**